### DISCUSSION TITLE
How use extensions  to change option default setting?

### DISCUSSION THREAD

**Discussion by 000yuying** [awcodes/filament-tiptap-editor#191] [Q&A] [answered]
Hello,
I have read the part about Custom Extensions in README.md. It seems to be more about adding new extensions. What should I do if I just want to change the existing default values?

There are some parts I want to change the settings but I don't know how to write them in Extensions.

1. To make highlight multicolored:
   Official documentation example:

   Highlight.configure({
     multicolor: true,
   })

2. To customize headings to display specific levels, for example, only display (H1, H2):
   Official documentation example:

   Heading.configure({
     levels: [1, 2],
   })

Additional question:
When I embed links to Instagram posts, most of the HTML content gets filtered out, making it impossible to display the post content. Do you have any suggestions on how to handle this?

**Comment by awcodes** [comment, marked as answer]
1 & 2. You would need to create custom extensions for them as well. And register them. If they have the same name / key as tiptap editor then it will replace the default ones.

For you Instagram question, you would need to create a custom tiptap extension to be able to do that. Tiptap and Prosemirror are very strict about what content can be placed in the editor, so you have to tell it was is acceptable in that situation.
